rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,199 @@
+
+This is a small portion of the total amount of interface we have to stabilize, and that will hopefully yield a narrow and more focused discussion.
+Yet it will still offer some interesting topics to discus, such as:
+
+- logging
+- store paths on `stdout` at end of build?
+- Should commands like `show-derivation` should use `--json` by default
+- Flat vs hierarchical commands
+- is `--derivation` a good flag?
+
+## Step 3: Stabilize the rest of the CLI, without Flakes
+
+If the deadline for step 1 is *not* met, we jump directly to this step, skipping step 2.
+
+This is rest off the new CLI, without flakes.
+Unlike the store-only Nix command which has yet to be implemented, this is easy to visual today by enabling the `nix-command` feature without the `flakes` feature.
+This is a chance to discuss topics like:
+
+- Is `--file` good, or should we be able to specify something like `file#attribute` to mix files and attributes? (Without using Flakes.)
+- Should all outputs be selected if one writes `foo.dev`?
+- How can `nix repl` have a more normal CLI?
+
+## Step 4: Stabilize Flakes itself
+
+Finally, with the other less controversial interfaces stabilized, we can tackle the Flakes itself, the one remainder.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Having laid out the plan, let us now return to how the current situation is characterized and see if the various facts that the factions orient themselves are respected.
+
+## Flakes are very popular
+
+And they will be stabilized, with minimal delay.
+
+Firstly, there is no feature work proposed in the interim --- the splitting of Nix is just partitioning existing functionality, with some behind-the-scenes refactors needed to make that possible.
+The deadline placed on that work also assuages those concerned about the current "limbo" that stabilization will be gotten to no matter what, even if the splitting fails, after a set amount of time.
+
+## Difficulties in the roll-out
+
+Process concerns are addressed by having a clear process, with clear outcomes, before any stabilization is begun.
+That is the purpose of this RFC!
+
+## Flakes are criticized for eating other Nix features
+
+We do *not* propose adding delay to once-again separate those features from Flakes, as that would introduce more feature work which would delay stabilization and be unacceptable to the pro-Flakes faction.
+
+That said, an ancillary incremental process of incremental stabilization is to bolster a *sense* of layering in Nix's user interface that has been, according to this camp, lost.
+Ensuring that the these two subsets of the new CLI --- without Flakes, and without Nix language support --- do in fact make sense in isolation will provide a "scaffolding" upon which interested parties can later introduce generalized features like search and pure eval without Flakes.
```

Comment:
```suggestion
Ensuring that these two subsets of the new CLI --- without Flakes, and without Nix language support --- do in fact make sense in isolation will provide a "scaffolding" upon which interested parties can later introduce generalized features like search and pure eval without Flakes.
```

Reply:
Can you clarify what the "without Nix language support" subset of the new CLI is referring to?

Reply:
The would be the store-only CLI.